builder/comp-builder.nix

```
@@ -289,7 +290,7 @@ let
             lib.optionalString (cabal-generator == "hpack") ''
               ${buildPackages.haskell-nix.internal-nix-tools}/bin/hpack
             ''
-        );
+        ) + lib.optionalString (prePatch != null) "\n${prePatch}";
```

```suggestion
        ) + lib.optionalString (prePatch != null) prePatch;
```
Why not this? if prePatch != null => prePatch is a string? And the previous items have line breaks at the end anyway?

---

> previous items have line breaks at the end anyway?

Sadly not and unfortunately I think fixing that would require a full rebuild.  See https://github.com/input-output-hk/haskell.nix/blob/f9d847393242854e261c480630569864753ead01/builder/comp-builder.nix#L286